client/keystore/src/local.rs

```diff
@@ -277,9 +277,13 @@ impl SyncCryptoStore for LocalKeystore {
 		self.0.write().insert_unknown(key_type, suri, public).map_err(|_| ())
 	}
 
-	fn has_keys(&self, public_keys: &[(Vec<u8>, KeyTypeId)]) -> bool {
-		public_keys.iter()
-			.all(|(p, t)| self.0.read().key_phrase_by_type(&p, *t).ok().flatten().is_some())
+	fn has_keys(&self, public_keys: &[(Vec<u8>, KeyTypeId)]) -> Vec<usize> {
+		public_keys.iter().enumerate().fold(Vec::new(), |mut idxs, (i, (p, t))| {
+			if self.0.read().key_phrase_by_type(&p, *t).ok().flatten().is_some() {
+				idxs.push(i);
+			}
+			idxs
```

Comment:
but this is changing the definition, before it requires all key to exist, now it requires only 1 key to exist.

I'm not very aware of keystore but this seems quite a different behavior to me

Reply:
Indeed, but I think that's what @bkchr wanted?